### PR TITLE
MueLu: Adding capability to do prolongator scaling by aggregate size

### PR DESCRIPTION
**Pull request by csiefer2**
Testing: New test added to TentativeP unit tests
Auto-PR for SHA 7086f41

### CODE REVIEW

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp [hunk 1]
```diff
@@ -496,7 +504,9 @@ namespace MueLu {
 
             for (size_t k = 0, lnnz = 0; k < NSDim; k++) {
               // Skip zeros (there may be plenty of them, i.e., NSDim > 1 or boundary conditions)
-              const SC qr_jk = fineNS[k][aggToRowMapLO[aggStart[agg]+j]];
+              SC qr_jk = fineNS[k][aggToRowMapLO[aggStart[agg]+j]];
+              if(constantColSums) qr_jk = qr_jk / (double)aggSizes[agg];
+                fineNS[k][aggToRowMapLO[aggStart[agg]+j]];
```
Comment (rstumin) [inline packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp]: This line seems like it does a NOP ... I'm guessing it is a cut-and-paste leftover?

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp [hunk 2]
```diff
@@ -520,7 +530,8 @@ namespace MueLu {
 
             for (size_t k = 0, lnnz = 0; k < NSDim; k++) {
               // Skip zeros (there may be plenty of them, i.e., NSDim > 1 or boundary conditions)
-              const SC qr_jk = fineNS[k][rowMap->getLocalElement(aggToRowMapGO[aggStart[agg]+j])];
+              SC qr_jk = fineNS[k][rowMap->getLocalElement(aggToRowMapGO[aggStart[agg]+j])];
+              if(constantColSums) qr_jk = qr_jk / (double)aggSizes[agg];
```
Comment (rstumin) [inline packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp]: Having all these if's is a little bit annoying from a performance perspective. Is it easy to duplicate loops ... which of course isn't great from a maintenance perspective.

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp]: Good catch.

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp]: One would hope that branch prediction --- of a const --- would do the right thing here.

Comment (rstumin) [reply packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_TentativePFactory_def.hpp]: Yeah ... I agree.

**Comment by csiefer2** [comment]
Gold files are my nemesis.

**Pull request merged**